lib/classes/ConfigSchemaHandler/normalizeAjvErrors.test.js

```diff
@@ -0,0 +1,563 @@
+'use strict';
+const Ajv = require('ajv');
+
+const expect = require('chai').expect;
+const normalizeAjvErrors = require('./normalizeAjvErrors');
+
+describe('#normalizeAjvErrors', () => {
+  const ajv = new Ajv({ allErrors: true });
+  let schema;
+  let userConfig;
+  let ajvErrors;
+
+  it('should show error messages for root and second level property', () => {
+    userConfig = {
+      foo: 'bar',
+      package: { incclude: ['./folder'] },
+    };
+    schema = {
+      type: 'object',
+      properties: {
+        package: {
+          type: 'object',
+          properties: {
+            include: { type: 'array', items: { type: 'string' } },
+          },
+          additionalProperties: false,
+        },
+      },
+      additionalProperties: false,
+    };
+    const validate = ajv.compile(schema);
+    validate(userConfig);
+    ajvErrors = validate.errors;
+    expect(normalizeAjvErrors(ajvErrors, userConfig).map(err => err.message)).to.deep.equal([
+      "Unrecognized property 'foo' on 'root'",
+      "Unrecognized property 'incclude' on 'package'",
+    ]);
+  });
+
+  it('should show error messages for function events related config ', () => {
+    userConfig = {
+      functions: {
+        someFunc: {
+          events: [
+            { httpp: 'get /foo' },
+            { http: 'gets /foo' },
+            {
+              http: {
+                path: '/home',
+                method: 'get',
+                invalidProp: 'baz',
+              },
+            },
+            {
+              http: {
+                method: 'get',
+              },
+            },
+            {
+              http: {
+                method: 'gets',
+                path: '/home',
+              },
+            },
+            {
+              http: null,
+              path: '/home',
+              method: 'get',
+            },
+          ],
+        },
+      },
+    };
+
+    const expectedErrorMessages = [
+      "Unsupported function event 'httpp' at functions.someFunc.events[0]",
+      'functions.someFunc.events[1].http should match pattern "^(get|post|put) [a-zA-Z0-9]+$"',
+      "Unrecognized property 'invalidProp' on 'functions.someFunc.events[2].http'",
+      "functions.someFunc.events[3].http should have required property 'path'",
+
+      // TODO: add list of allowed values as AJV error contains this list in error object
+      'functions.someFunc.events[4].http.method should be equal to one of the allowed values',
```

Comment:
Is this a challenge to list those values? (I would do it, but only in n generic way, so for all errors of this type, we can tweak message so it contains also list of accepted values)

Reply:
I'll add it